src/utils/dayViewLayout.js

```diff
@@ -355,10 +358,56 @@ export function getStyledAvailabilities ({
     totalMin,
   };
 
-  availabilities.forEach((availability, availabilityIdx) => {
-    let style = getYStyles(availabilityIdx, helperArgs);
-    styledAvailabilities.push({ availability, style });
-  })
+  const overlapMap = {};
+  let columnIndex = 0;
+
+  for (let availabilityIndex = 0; availabilityIndex < availabilities.length; availabilityIndex++) {
+    const availability = availabilities[availabilityIndex];
+    let grouped = false;
+
+    // Check for overlap with existing groups
+    const overlapKeys = Object.keys(overlapMap);
+    for (let groupIndex = 0; groupIndex < overlapKeys.length; groupIndex++) {
+      const groupKey = overlapKeys[groupIndex];
+      const group = overlapMap[groupKey];
+      const lastAvailability = group[group.length - 1];
+      const lastAvailabilityEndTime = get(lastAvailability, availabilityEndAccessor);
+      const availabilityStartTime = get(availability, availabilityStartAccessor);
+      if (lastAvailabilityEndTime <= availabilityStartTime) {
+        // No overlap, add to current group
+        group.push(availability);
+        grouped = true;
```

Comment:
Thoughts on using a name like `placed` instead, to indicate that the availability has been placed into an overlap-column? `grouped` was confusing to me because I thought it was synonymous with there being an overlap (e.g. overlapping appointments on the calendar form a visual "group") but it's actually the opposite.